packages/jsii-pacmak/lib/targets/dotnet/dotnetgenerator.ts

```diff
@@ -933,6 +1115,21 @@ export class DotNetGenerator extends Generator {
     const access = this.renderAccessLevel(prop);
     const staticKeyWord = prop.static ? 'static ' : '';
     const propName = this.nameutils.convertPropertyName(prop.name);
+    const propTypeFQN = this.typeresolver.toDotNetType(prop.type);
+    const isOptional = prop.optional ? '?' : '';
+
+    // We need to use a backing field so we can perform type checking if the property type is a union, and this is a struct.
```

Comment:
why do we need backing fields here? why does type checking a union type property on a struct require a backing field?